ui/controller/controller.go

```diff
@@ -662,6 +678,44 @@ func (c *Controller) SetTrackRatings(trackIDs []string, rating int) {
 	}
 }
 
+func (c *Controller) CreateShareURL(id string) {
+	go func() {
+		r, ok := c.App.ServerManager.Server.(mediaprovider.SupportsSharing)
+		if !ok {
+			return
+		}
+
+		shareUrl, err := r.CreateShareURL(id)
+		if err != nil {
+			log.Printf("error creating share URL: %v", err)
```

Comment:
As mentioned in Discord, let's notify the user of the error here using `dialog.ShowError`, since I think this will be a super common error encountered (Navidrome disables sharing by default). Also add a TODO comment to switch over to an error toast, once a toast mechanism is available.

Reply:
Good suggestion. I have added a `showError` method in the controller, so we can simply refactor that implementation to use a toast instead.